cmake/modules/hunter_setup_msvc.cmake

```diff
@@ -115,32 +115,45 @@ macro(hunter_setup_msvc)
     hunter_status_debug(
         "CMAKE_VS_DEVENV_COMMAND: '${CMAKE_VS_DEVENV_COMMAND}'"
     )
+    hunter_status_debug(
+        "CMAKE_VS_MSBUILD_COMMAND: '${CMAKE_VS_MSBUILD_COMMAND}'"
+    )
 
     string(COMPARE EQUAL "${_hunter_vcvarsall_path}" "" _is_empty)
     if(_is_empty)
       if(HUNTER_TESTING)
         # ignore error, see 'tests/hunter_setup_msvc/CMakeLists.txt'
       else()
         hunter_status_debug(
-            "Environment variable '${_hunter_vcvarsall_env}' is empty, analyzing CMAKE_VS_DEVENV_COMMAND"
+            "Environment variable '${_hunter_vcvarsall_env}' is empty,"
+            "  analyzing CMAKE_VS_DEVENV_COMMAND and CMAKE_VS_MSBUILD_COMMAND"
         )
-        string(COMPARE EQUAL "${CMAKE_VS_DEVENV_COMMAND}" "" is_empty)
-        if(is_empty)
-          hunter_fatal_error(
-              "Incorrect CMAKE_VS_DEVENV_COMMAND: is empty"
-              WIKI
-              error.vs.devenv
+        string(COMPARE EQUAL "${CMAKE_VS_DEVENV_COMMAND}" "" is_devenv_empty)
+        string(COMPARE EQUAL "${CMAKE_VS_MSBUILD_COMMAND}" "" is_msbuild_empty)
+        if(NOT is_devenv_empty AND IS_ABSOLUTE "${CMAKE_VS_DEVENV_COMMAND}")
+          get_filename_component(_hunter_vcvarsall_path
+              "${CMAKE_VS_DEVENV_COMMAND}" DIRECTORY
           )
-        endif()
-        if(NOT IS_ABSOLUTE "${CMAKE_VS_DEVENV_COMMAND}")
+          set(_hunter_vcvarsall_path
+              "${_hunter_vcvarsall_path}/../../VC/Auxiliary/Build"
+          )
+        elseif(is_msbuild_empty AND IS_ABSOLUTE "${CMAKE_VS_MSBUILD_COMMAND}")
```

Comment:
`NOT` is missing? It should be
```cmake
elseif(NOT is_msbuild_empty AND IS_ABSOLUTE "${CMAKE_VS_MSBUILD_COMMAND}")
```
?

Reply:
Yes I think it should match the Devenv case. Sorry about that.

Reply:
Fix: https://github.com/ruslo/hunter/commit/556dcfea6b4b8051404de4eaac897ef5086c96e0

Released: https://github.com/ruslo/hunter/releases/tag/v0.23.110

> Yes I think it should match the Devenv case. Sorry about that.

> we can use the path to MSBuild to find it

Well, I was thinking it's exactly the branch that do the fix for you. Have you tested it?

`if(NOT is_devenv_empty AND IS_ABSOLUTE "${CMAKE_VS_DEVENV_COMMAND}")` branch used on my local machine and in AppVeyor testing, that's why bug wasn't caught.

Reply:
You are right. I negated the check to check the error message case, then accidentally pushed the result. I have sent you a beer via PayPal by way of apology.